packages/core/admin/shared/contracts/authentication.ts

```diff
@@ -0,0 +1,118 @@
+import type { AdminUser, SanitizedAdminUser } from './shared';
+
+/**
+ * /login - Log in as an admin user
+ */
+export declare namespace Login {
+  export interface Request {
+    query: {
+      user: Pick<AdminUser, 'email' | 'password'>;
+    };
+  }
+
+  export interface Response {
+    data: {
+      token: string;
+      user: Omit<SanitizedAdminUser, 'permissions'>;
+    };
+  }
+}
```

Comment:
@jhoward1994 @gu-stav @joshuaellis wdyt of this?

It's using both modules and namespaces, and in my opinion has the best of both worlds:
- Contracts are encapsulated per logical groups (authentication endpoints in here)
- Every contract will always have a Request and Response interface

```
type RegisterBody = Register.Request['body'];
```

Reply:
yes

Reply:
looks good to me 👍🏻